devito/operator.py

```diff
@@ -518,6 +515,21 @@ def _cglobals(self):
 # Misc helpers
 
 
+def set_dse_mode(mode):
+    """
+    Transform :class:`Operator` input in a format understandable by the DLE.
+    """
+    if not mode:
+        return 'noop'
+    elif isinstance(mode, str):
+        return mode
+    else:
+        try:
+            return ','.join(mode)
+        except:
+            raise TypeError("Illegal DSE mode %s." % str(mode))
+
+
```

Comment:
the `Operator` decides if/how/when to use the DSE/DLE, potentially one may write an Operator that defaults DSE/DLE modes to `basic` or whatever desired, regardless of what the user specifies